src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.component.js

```diff
@@ -58,60 +74,146 @@ const Index = ({
             }
         };
 
-        return selectedOptionId && availableSearchOptions[selectedOptionId].searchGroups
-            .filter(searchGroup => searchGroup.unique)
-            .map(({ searchForm, formId, searchScope }) => {
-                const name = searchForm.getElements()[0].formName;
-                return (
-                    <Section
-                        className={classes.searchDomainSelectorSection}
-                        header={
-                            <SectionHeaderSimple
-                                containerStyle={{ paddingLeft: 8, borderBottom: '1px solid #ECEFF1' }}
-                                title={i18n.t('Search {{name}}', { name })}
-                            />
-                        }
-                    >
-                        <div className={classes.searchRow}>
-                            <div className={classes.searchRowSelectElement}>
-                                {
-                                    forms[formId] &&
-                                    <Form
-                                        formRef={
-                                            (formInstance) => { formReference[formId] = formInstance; }
+        const handleSearchViaAttributes = (selectedId, formId, searchScope, minAttributesRequiredToSearch) => {
+            const isValid = isSearchViaAttributesValid(minAttributesRequiredToSearch, formId);
+
+            if (isValid) {
+                setError(false);
+                switch (searchScope) {
+                case searchScopes.PROGRAM:
+                    searchViaAttributesOnScopeProgram({ programId: selectedId, formId });
+                    break;
+                case searchScopes.TRACKED_ENTITY_TYPE:
+                    searchViaAttributesOnScopeTrackedEntityType({ trackedEntityTypeId: selectedId, formId });
+                    break;
+                default:
+                    break;
+                }
+            } else {
+                setError(true);
+            }
+        };
+
+        return (<>
+            {
+                selectedOptionId && availableSearchOptions[selectedOptionId].searchGroups
+                    .filter(searchGroup => searchGroup.unique)
```

Comment:
Renders only the unique related fields. I am doing that because to me it looks that validation is being handled differently between unique and not unique search groups.